burst-kotlin-plugin/src/main/kotlin/app/cash/burst/kotlin/InterceptorInjector.kt

```diff
@@ -406,6 +405,86 @@ internal class InterceptorInjector(
     original.patchDeclarationParents()
   }
 
+  private fun IrBlockBodyBuilder.createTestFunctionClass(
+    nameHint: String,
+    packageName: IrValueDeclaration,
+    className: IrValueDeclaration,
+    functionName: IrValueDeclaration,
+    proceed: IrExpression,
+  ): IrClass {
+    return createTestFunctionClass(
+      nameHint = nameHint,
+      packageName = irGet(packageName),
+      className = irGet(className),
+      functionName = irGet(functionName),
+    ) {
+      irFunctionBody(
+        context = context,
+        scopeOwnerSymbol = scope.scopeOwnerSymbol,
+      ) {
+        +proceed
+      }
+    }
+  }
+
+  /** Create a subclass of `TestFunction` and returns its constructor. */
+  private fun createTestFunctionClass(
```

Comment:
It’s a decent amount more IR this way, but it should be about the same amount of bytecode. Plus we get more control over our symbol names.